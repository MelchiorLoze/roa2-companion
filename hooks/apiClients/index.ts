export { useGameApiClient } from './useGameApiClient/useGameApiClient';
