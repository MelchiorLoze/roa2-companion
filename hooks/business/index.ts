export { useAuth } from './useAuth/useAuth';
export { useCurrencyBalance } from './useCurrencyBalance/useCurrencyBalance';
export { useLeaderboardStats } from './useLeaderboardStats/useLeaderboardStats';
export { useRotatingCoinShop } from './useRotatingCoinShop/useRotatingCoinShop';
export { useUserStats } from './useUserStats/useUserStats';
