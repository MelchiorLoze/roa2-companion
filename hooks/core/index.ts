export { useStorageState } from '../core/useStorageState/useStorageState';
export { useAppFonts } from './useAppFonts/useAppFonts';
export { useHttpClient } from './useHttpClient/useHttpClient';
