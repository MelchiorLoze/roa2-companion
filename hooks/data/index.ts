export { useGetEntityToken } from './useGetEntityToken/useGetEntityToken';
export { useGetInventoryItems } from './useGetInventoryItems/useGetInventoryItems';
export { useGetItems } from './useGetItems/useGetItems';
export { useGetPlayerStatistics } from './useGetPlayerStatistics/useGetPlayerStatistics';
export { useGetUserReadOnlyData } from './useGetUserReadOnlyData/useGetUserReadOnlyData';
export { useLoginWithEmail } from './useLoginWithEmail/useLoginWithEmail';
export { useGetMyRotationalCoinStore } from './useMyRotationalCoinStore/useGetMyRotationalCoinStore';
