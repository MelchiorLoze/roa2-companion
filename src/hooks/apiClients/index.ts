export { useGameApiClient } from './useGameApiClient/useGameApiClient';
export { useSteamCommunityApiClient } from './useSteamCommunityApiClient/useSteamCommunityApiClient';
