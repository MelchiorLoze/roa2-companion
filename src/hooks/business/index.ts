export { useAuth } from './useAuth/useAuth';
export { useCurrencyBalance } from './useCurrencyBalance/useCurrencyBalance';
export { useLeaderboardStats } from './useLeaderboardStats/useLeaderboardStats';
export { useUserStats } from './useUserStats/useUserStats';
