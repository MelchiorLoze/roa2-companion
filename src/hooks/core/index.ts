export { useAppFonts } from './useAppFonts/useAppFonts';
export { useHttpClient } from './useHttpClient/useHttpClient';
export { useStorageState } from './useStorageState/useStorageState';
