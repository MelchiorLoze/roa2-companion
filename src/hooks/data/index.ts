// GAME API
export { useGetEntityToken } from './useGetEntityToken/useGetEntityToken';
export { useGetInventoryItems } from './useGetInventoryItems/useGetInventoryItems';
export { useGetLeaderboardAroundPlayer } from './useGetLeaderboardAroundPlayer/useGetLeaderboardAroundPlayer';
export { useGetPlayerStatistics } from './useGetPlayerStatistics/useGetPlayerStatistics';
export { useGetUserReadOnlyData } from './useGetUserReadOnlyData/useGetUserReadOnlyData';
export { useLoginWithEmail } from './useLoginWithEmail/useLoginWithEmail';
export { useSendAccountRecoveryEmail } from './useSendAccountRecoveryEmail/useSendAccountRecoveryEmail';

// STEAM API
export { useCommunityLeaderboard } from './useCommunityLeaderboard/useCommunityLeaderboard';
export { useCommunityLeaderboards } from './useCommunityLeaderboards/useCommunityLeaderboards';
