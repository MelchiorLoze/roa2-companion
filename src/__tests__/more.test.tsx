import { fireEvent, render, screen } from '@testing-library/react-native';
import { useRouter } from 'expo-router';

import More from '@/app/(private)/more';
import { useAuth } from '@/hooks/business';

jest.mock('expo-router');
const useRouterMock = jest.mocked(useRouter);
const navigateMock = jest.fn();
useRouterMock.mockReturnValue({
  navigate: navigateMock,
} as unknown as ReturnType<typeof useRouter>);

jest.mock('../hooks/business');
const useAuthMock = jest.mocked(useAuth);
const logoutMock = jest.fn();
useAuthMock.mockReturnValue({
  isLoggedIn: true,
  login: jest.fn(),
  logout: logoutMock,
  isLoading: false,
  isError: false,
});

const renderComponent = () => {
  const result = render(<More />);

  expect(useAuthMock).toHaveBeenCalledTimes(1);
  expect(logoutMock).not.toHaveBeenCalled();

  return result;
};

describe('More', () => {
  it('matches the snapshot', () => {
    const tree = renderComponent().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the list of external links', () => {
    renderComponent();

    const links = screen.getAllByRole('link');

    expect(links).toHaveLength(6);
  });

  it('navigates to about page on button press', () => {
    renderComponent();

    const button = screen.getByRole('button', { name: 'About this app' });
    fireEvent.press(button);

    expect(navigateMock).toHaveBeenCalledTimes(1);
    expect(navigateMock).toHaveBeenCalledWith('/about');
  });

  it('calls logout on button press', () => {
    renderComponent();

    const button = screen.getByRole('button', { name: 'Log out' });
    fireEvent.press(button);

    expect(logoutMock).toHaveBeenCalledTimes(1);
  });
});
