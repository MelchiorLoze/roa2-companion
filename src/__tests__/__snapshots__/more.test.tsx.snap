// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`More matches the snapshot 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        {
          "gap": 24,
        }
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="link"
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#0E0B2A",
                "flexDirection": "row",
                "gap": 8,
                "padding": 12,
                "paddingHorizontal": 16,
              },
              false,
            ]
          }
        >
          <ViewManagerAdapter_ExpoImage
            containerViewRef={"[React.ref]"}
            contentFit="contain"
            contentPosition={
              {
                "left": "50%",
                "top": "50%",
              }
            }
            height={20}
            nativeViewRef={"[React.ref]"}
            onError={[Function]}
            onLoad={[Function]}
            onLoadStart={[Function]}
            onProgress={[Function]}
            placeholder={[]}
            source={
              [
                {
                  "uri": "https://dragdown.wiki/favicon.ico",
                },
              ]
            }
            style={
              {
                "height": 20,
                "width": 20,
              }
            }
            transition={null}
            width={20}
          />
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 18,
                  "textTransform": "uppercase",
                },
                false,
              ]
            }
          >
            Game wiki
          </Text>
          <
            color="white"
            name="arrow-outward"
            size={20}
            style={
              {
                "marginLeft": "auto",
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="link"
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#0E0B2A",
                "flexDirection": "row",
                "gap": 8,
                "padding": 12,
                "paddingHorizontal": 16,
              },
              false,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 18,
                  "textTransform": "uppercase",
                },
                false,
              ]
            }
          >
            Frame data
          </Text>
          <
            color="white"
            name="arrow-outward"
            size={20}
            style={
              {
                "marginLeft": "auto",
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="link"
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#0E0B2A",
                "flexDirection": "row",
                "gap": 8,
                "padding": 12,
                "paddingHorizontal": 16,
              },
              false,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 18,
                  "textTransform": "uppercase",
                },
                false,
              ]
            }
          >
            Elo History
          </Text>
          <
            color="white"
            name="arrow-outward"
            size={20}
            style={
              {
                "marginLeft": "auto",
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="link"
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#0E0B2A",
                "flexDirection": "row",
                "gap": 8,
                "padding": 12,
                "paddingHorizontal": 16,
              },
              false,
            ]
          }
        >
          <ViewManagerAdapter_ExpoImage
            containerViewRef={"[React.ref]"}
            contentFit="contain"
            contentPosition={
              {
                "left": "50%",
                "top": "50%",
              }
            }
            height={20}
            nativeViewRef={"[React.ref]"}
            onError={[Function]}
            onLoad={[Function]}
            onLoadStart={[Function]}
            onProgress={[Function]}
            placeholder={[]}
            source={
              [
                {
                  "uri": "https://www.redditstatic.com/shreddit/assets/favicon/64x64.png",
                },
              ]
            }
            style={
              {
                "height": 20,
                "width": 20,
              }
            }
            transition={null}
            width={20}
          />
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 18,
                  "textTransform": "uppercase",
                },
                false,
              ]
            }
          >
            Game Reddit
          </Text>
          <
            color="white"
            name="arrow-outward"
            size={20}
            style={
              {
                "marginLeft": "auto",
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="link"
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#0E0B2A",
                "flexDirection": "row",
                "gap": 8,
                "padding": 12,
                "paddingHorizontal": 16,
              },
              false,
            ]
          }
        >
          <ViewManagerAdapter_ExpoImage
            containerViewRef={"[React.ref]"}
            contentFit="contain"
            contentPosition={
              {
                "left": "50%",
                "top": "50%",
              }
            }
            height={20}
            nativeViewRef={"[React.ref]"}
            onError={[Function]}
            onLoad={[Function]}
            onLoadStart={[Function]}
            onProgress={[Function]}
            placeholder={[]}
            source={
              [
                {
                  "uri": "https://cdn.prod.website-files.com/6257adef93867e50d84d30e2/62fddf0fde45a8baedcc7ee5_847541504914fd33810e70a0ea73177e%20(2)-1.png",
                },
              ]
            }
            style={
              {
                "height": 20,
                "width": 20,
              }
            }
            transition={null}
            width={20}
          />
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 18,
                  "textTransform": "uppercase",
                },
                false,
              ]
            }
          >
            Game Discord
          </Text>
          <
            color="white"
            name="arrow-outward"
            size={20}
            style={
              {
                "marginLeft": "auto",
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="button"
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#0E0B2A",
                "flexDirection": "row",
                "gap": 8,
                "padding": 12,
                "paddingHorizontal": 16,
              },
              false,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 18,
                  "textTransform": "uppercase",
                },
                false,
              ]
            }
          >
            About this app
          </Text>
          <
            color="white"
            name="arrow-forward"
            size={20}
            style={
              {
                "marginLeft": "auto",
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="button"
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#0E0B2A",
                "flexDirection": "row",
                "gap": 8,
                "padding": 12,
                "paddingHorizontal": 16,
              },
              false,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "FranklinGothicDemiCond-Regular",
                  "fontSize": 18,
                  "textTransform": "uppercase",
                },
                false,
              ]
            }
          >
            Log out
          </Text>
          <
            color="white"
            name="logout"
            size={20}
            style={
              {
                "marginLeft": "auto",
              }
            }
          />
        </View>
        <View
          style={
            {
              "backgroundColor": "#CACEFF",
              "height": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
